applications/FluidDynamicsApplication/custom_utilities/statistics_record.cpp

```diff
@@ -0,0 +1,158 @@
+#include "includes/define.h"
+#include "includes/element.h"
+#include "containers/variable.h"
+#include "utilities/openmp_utils.h"
+
+#include "statistics_record.h"
+#include "statistics_data.h"
+
+#include "fluid_dynamics_application_variables.h"
+
+namespace Kratos {
+
+void StatisticsRecord::AddResult(StatisticsSampler::Pointer pResult)
+{
+    KRATOS_TRY
+
+    KRATOS_ERROR_IF(mInitialized) << "Trying to add statistical data after Initialization of the internal storage." << std::endl;
+
+    std::size_t result_size = pResult->GetSize();
+    pResult->SetOffset(mDataBufferSize);
+
+    mDataBufferSize += result_size;
+    mAverageData.push_back(pResult);
+
+    KRATOS_CATCH("")
+}
+
+void StatisticsRecord::AddHigherOrderStatistic(StatisticsSampler::Pointer pResult)
+{
+    KRATOS_TRY
+
+    KRATOS_ERROR_IF(mInitialized) << "Trying to add statistical data after Initialization of the internal storage." << std::endl;
+
+    std::size_t result_size = pResult->GetSize();
+    pResult->SetOffset(mDataBufferSize);
+
+    mDataBufferSize += result_size;
+    mHigherOrderData.push_back(pResult);
+
+    KRATOS_CATCH("")
+}
+
+void StatisticsRecord::InitializeStorage(ModelPart::ElementsContainerType& rElements)
+{
+    mUpdateBuffer.resize(OpenMPUtils::GetNumThreads());
+    #pragma omp parallel
+    {
+        unsigned int k = OpenMPUtils::ThisThread();
```

Comment:
ok here, but we should try to avoid using ThisThread to be futureproof (not a complaint here specifically)

Reply:
I agree, but I understand that the clean alternative here is to declare the mUpdateBuffer as a threadprivate static member, which is unfortunately OpenMP 3.0 and does not compile in Windows.